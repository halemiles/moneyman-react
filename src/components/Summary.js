import React, { useState, useEffect } from "react";
import Form from 'react-bootstrap/Form';

import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';

import 'bootstrap/dist/css/bootstrap.min.css';

export default function Summary(props) {

    const [actual, setActual] = useState(2200);
    const [due, setDue] = useState(0);
    const [remaining, setRemaining] = useState(0);

    useEffect(() => {
      let totalDue = 0;
      props.planDates.forEach((pd) => {
        totalDue += pd.amount;
      });
      setDue(totalDue.toFixed(2));
      setRemaining((actual - totalDue).toFixed(2));
    }, [props.planDates, actual]);

    const handleActualChange = (e) => {
      setActual(e.target.value);
    };



  return (
    <div>
      <h2>Summary</h2>
      <div>
            <Form>
              <Form.Group controlId="actualValue" as={Row} className="mb-3">
                <Form.Label column sm="2">Actual:</Form.Label>
                <Col sm="8">
                  <Form.Control type="number" step="0.01" value={actual} onChange={handleActualChange} />
                </Col>
            </Form.Group>
            </Form>
        <p>Due: {due}</p>
        <p>Remaining: {remaining}</p>
      </div>
    </div>
  );

  
}