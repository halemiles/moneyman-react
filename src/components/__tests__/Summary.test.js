import React from 'react';
import { render, screen } from '@testing-library/react';
import Summary from '../Summary';

// mock props. plandates should contains data
const props = {
    planDates: [
        {
            transactionName: "transaction 1",
            amount: 10,
            date: "2021-01-01"
        },
        {
            transactionName: "trandaction 2",
            amount: 90,
            date: "2021-01-01"
        }
    ]
};

test('Summary should contain actual', () => {
    render(<Summary {...props} />);
    const linkElement = screen.getByText(/Actual/i);
    expect(linkElement).toBeInTheDocument();
});

// input with id actualValue should contain 2200
test('Actual input should contain 2200', () => {
    render(<Summary {...props} />);
    const linkElement = screen.getByDisplayValue("2200");
    expect(linkElement).toBeInTheDocument();
}
);

test('Due should contain 100', () => {
    render(<Summary {...props} />);
    const linkElement = screen.getByText("Due: 100.00");
    expect(linkElement).toBeInTheDocument();
}
);

// snapshot test
test('Summary should match snapshot', () => {
    const { container } = render(<Summary {...props} />);
    expect(container).toMatchSnapshot();
});


//  test('Summary should contain actual', () => {
//     render(<Summary />);
//     const linkElement = screen.getByText(/Actual/i);
//     expect(linkElement).toBeInTheDocument();    
//  });

//  test('Summary should contain due', () => {
//     render(<Summary />);
//     const linkElement = screen.getByText(/Due/i);
//     expect(linkElement).toBeInTheDocument();    
//  });

//  test('Summary should contain remaining', () => {
//     render(<Summary />);
//     const linkElement = screen.getByText(/Remaining/i);
//     expect(linkElement).toBeInTheDocument();    
//  });