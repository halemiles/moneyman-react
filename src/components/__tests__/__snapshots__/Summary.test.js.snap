// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should match snapshot 1`] = `
<div>
  <div>
    <h2>
      Summary
    </h2>
    <div>
      <form
        class=""
      >
        <div
          class="mb-3 row"
        >
          <label
            class="form-label col-form-label col-sm-2"
            for="actualValue"
          >
            Actual:
          </label>
          <div
            class="col-sm-8"
          >
            <input
              class="form-control"
              id="actualValue"
              step="0.01"
              type="number"
              value="2200"
            />
          </div>
        </div>
      </form>
      <p>
        Due: 
        100.00
      </p>
      <p>
        Remaining: 
        2100.00
      </p>
    </div>
  </div>
</div>
`;
